src/transactionComputer.ts

```diff
@@ -77,11 +82,21 @@ export class TransactionComputer {
     }
 
     applyGuardian(transaction: ITransaction, guardian: string) {
-        transaction.version = 2;
+        if (transaction.version < TRANSACTION_VERSION_DEFAULT) {
+            transaction.version = TRANSACTION_VERSION_DEFAULT;
+        }
```

Comment:
Tricky, but correct :+1: 

Since `TRANSACTION_VERSION_DEFAULT >= 2`.

Reply:
A separate constant called `TRANSACTION_MIN_VERSION_THAT_SUPPORTS_OPTIONS` would be more explicit. This is optional, however, just an idea / thinking out loud.

Reply:
added a const `TRANSACTION_MIN_VERSION_THAT_SUPPORTS_OPTIONS`.